examples/colour-uniform/main.rs

```diff
@@ -1712,11 +1625,74 @@ impl<B: Backend> Drop for FramebufferState<B> {
+    let window_builder = winit::window::WindowBuilder::new()
+        .with_min_inner_size(winit::dpi::LogicalSize::new(1.0, 1.0))
+        .with_inner_size(winit::dpi::LogicalSize::new(
+            DIMS.width as _,
+            DIMS.height as _,
+        ))
+        .with_title("colour-uniform".to_string());
+
+    let (backend, _instance) = create_backend(window_builder, &event_loop);
+
+    let mut renderer_state = unsafe { RendererState::new(backend) };
+
+    println!("\nInstructions:");
+    println!("\tChoose whether to change the (R)ed, (G)reen or (B)lue color by pressing the appropriate key.");
+    println!("\tType in the value you want to change it to, where 0 is nothing, 255 is normal and 510 is double, ect.");
+    println!("\tThen press C to change the (C)lear colour or (Enter) for the image color.");
+    println!(
+        "\tSet {:?} color to: {} (press enter/C to confirm)",
+        renderer_state.cur_color, renderer_state.cur_value
+    );
+
+    event_loop.run(move |event, _, control_flow| {
+        *control_flow = winit::event_loop::ControlFlow::Wait;
+
+        match event {
+            winit::event::Event::WindowEvent { event, .. } => {
+                #[allow(unused_variables)]
+                match event {
+                    winit::event::WindowEvent::KeyboardInput {
+                        input:
+                            winit::event::KeyboardInput {
+                                virtual_keycode: Some(winit::event::VirtualKeyCode::Escape),
+                                ..
+                            },
+                        ..
+                    }
+                    | winit::event::WindowEvent::CloseRequested =>
+                        *control_flow = winit::event_loop::ControlFlow::Exit,
+                    winit::event::WindowEvent::Resized(dims) => {
+                        #[cfg(feature = "gl")]
+                        renderer_state.backend.surface.get_window_t().resize(dims.to_physical(
+                            renderer_state.backend.surface.get_window_t().window().hidpi_factor(),
+                        ));
+                        renderer_state.recreate_swapchain = true;
+                    }
+                    winit::event::WindowEvent::KeyboardInput {
+                        input:
+                            winit::event::KeyboardInput {
+                                virtual_keycode,
+                                state: winit::event::ElementState::Pressed,
+                                ..
+                            },
+                        ..
+                    } => {
+                        if let Some(virtual_keycode) = virtual_keycode {
+                            renderer_state.input(virtual_keycode);
+                        }
+                    }
+                    _ => (),
+                }
+            }
+            winit::event::Event::EventsCleared => {
+                renderer_state.draw();
```

Comment:
You should be calling winit's `request_redraw` here.

`renderer_state.draw();` should be moved to `RedrawRequested`, so that it also redraws when the OS requests a redraw.

Reply:
Ah interesting, I missed RedrawRequested.
Why do we call request_redraw, shouldn't winit know when I need to redraw anyway?

Reply:
Well, winit will only issue a `RedrawRequested` when the OS tells it so. So, only if there has been a window resize, or parts of the window have been (unexposed then) exposed. However, simply receiving new input does not mean anything should redraw, which is why you should manually call `request_redraw`.

Reply:
So I'm thinking the quad example should not call `request_redraw` as it is a static image.
However the colour-uniform should call it at EventsCleared as it responds to user input.

Reply:
Yeah, that is about right.